astropy/io/fits/hdu/base.py

```diff
@@ -1464,55 +1464,23 @@ def _compute_checksum(self, data, sum32=0):
         -------
         ones complement checksum
         """
-        blocklen = 2880
-        sum32 = np.uint32(sum32)
-        for i in range(0, len(data), blocklen):
-            length = min(blocklen, len(data) - i)  # ????
-            sum32 = self._compute_hdu_checksum(data[i : i + length], sum32)
-        return sum32
-
-    def _compute_hdu_checksum(self, data, sum32=0):
-        """
-        Translated from FITS Checksum Proposal by Seaman, Pence, and Rots.
-        Use uint32 literals as a hedge against type promotion to int64.
-
-        This code should only be called with blocks of 2880 bytes
-        Longer blocks result in non-standard checksums with carry overflow
-        Historically,  this code *was* called with larger blocks and for that
-        reason still needs to be for backward compatibility.
-        """
-        u8 = np.uint32(8)
-        u16 = np.uint32(16)
-        uFFFF = np.uint32(0xFFFF)
-
-        if data.nbytes % 2:
-            last = data[-1]
-            data = data[:-1]
-        else:
-            last = np.uint32(0)
-
-        data = data.view(">u2")
-
-        hi = sum32 >> u16
-        lo = sum32 & uFFFF
```

Comment:
Oh, I see it now; I read this wrong (I'm really not used to hexadecimal) and thought that each `F` represented 8 bits when it's really 4, hence `& uFFFF` is only masking 16 bits instead of 32 as I assumed.
Thanks for the explanation though, it's really useful !